File: CiaiControllerSDK/Interfaces/ICommunication.cs
using System.Threading.Tasks;

namespace CiaiControllerSDK.Interfaces
{
    /// <summary>
    /// 通信接口基类
    /// </summary>
    public interface ICommunication
    {
        /// <summary>
        /// 是否已连接
        /// </summary>
        bool IsConnected { get; }

        #region 异步方法

        /// <summary>
        /// 连接设备
        /// </summary>
        Task<bool> ConnectAsync();

        /// <summary>
        /// 断开连接
        /// </summary>
        Task DisconnectAsync();

        /// <summary>
        /// 发送数据
        /// </summary>
        Task<bool> SendAsync(byte[] data);

        /// <summary>
        /// 接收数据
        /// </summary>
        Task<byte[]> ReceiveAsync();

        /// <summary>
        /// 发送并接收
        /// </summary>
        Task<byte[]> SendAndReceiveAsync(byte[] data);

        #endregion

        #region 同步方法

        /// <summary>
        /// 连接设备（同步）
        /// </summary>
        bool Connect();

        /// <summary>
        /// 断开连接（同步）
        /// </summary>
        void Disconnect();

        /// <summary>
        /// 发送数据（同步）
        /// </summary>
        bool Send(byte[] data);

        /// <summary>
        /// 接收数据（同步）
        /// </summary>
        byte[] Receive();

        /// <summary>
        /// 发送并接收（同步）
        /// </summary>
        byte[] SendAndReceive(byte[] data);

        #endregion
    }
}

File: CiaiControllerSDKForJava/src/main/java/com/ciai/controller/sdk/interface_/ICommunication.java
package com.ciai.controller.sdk.interface_;

import java.util.concurrent.CompletableFuture;

/**
 * 通信接口
 */
public interface ICommunication {

    /**
     * 是否已连接
     */
    boolean isConnected();

    /**
     * 异步连接
     */
    CompletableFuture<Boolean> connectAsync();

    /**
     * 异步断开连接
     */
    CompletableFuture<Void> disconnectAsync();

    /**
     * 异步发送数据
     */
    CompletableFuture<Boolean> sendAsync(byte[] data);

    /**
     * 异步接收数据
     */
    CompletableFuture<byte[]> receiveAsync();

    /**
     * 异步发送并接收数据
     */
    CompletableFuture<byte[]> sendAndReceiveAsync(byte[] data);

    /**
     * 同步连接
     */
    boolean connect();

    /**
     * 同步断开连接
     */
    void disconnect();

    /**
     * 同步发送数据
     */
    boolean send(byte[] data);

    /**
     * 同步接收数据
     */
    byte[] receive();

    /**
     * 同步发送并接收数据
     */
    byte[] sendAndReceive(byte[] data);
}
